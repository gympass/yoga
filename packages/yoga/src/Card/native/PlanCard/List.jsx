import React, { isValidElement } from 'react';
import styled, { withTheme } from 'styled-components';
import { TouchableWithoutFeedback } from 'react-native';
import { string, node, shape, oneOfType, func } from 'prop-types';
import get from 'lodash.get';

import Text from '../../../Text';
import Icon from '../../../Icon';
import theme from '../../../Theme/helpers/themeReader';

const { card, cardnative } = theme.components;

const List = styled.View`
  margin-top: ${cardnative.plan.list.margin.top}px;
`;

const Item = styled.View`
  margin-bottom: ${card.plan.list.item.margin.bottom}px;
`;

const Wrapper = styled.View`
  flex-direction: row;
  align-items: center;
`;

const IconWrapper = styled.View`
  margin-right: ${cardnative.plan.list.item.icon.margin.right}px;
`;

const Button = styled.View`
  margin-top: ${card.plan.list.button.margin.top}px;
`;

const ButtonText = styled(Text.Medium)`
  font-size: ${card.plan.list.button.font.size}px;
  color: ${card.plan.list.button.font.color};
`;

const ItemText = styled(Text.Small)`
  color: ${({ color }) => color || card.plan.list.item.font.color};
`;

const ListItem = withTheme(
  ({
    text,
    variant = undefined,
    theme: yogaTheme,
    icon = undefined,
    buttonProps: { children, ...buttonProps } = {},
  }) => {
    const itemColor = get(yogaTheme.yoga.colors, variant);

    if (variant && !itemColor)
      // eslint-disable-next-line no-console
      console.warn(
        `Invalid token ${variant}, you can use ${JSON.stringify(
          Object.keys(yogaTheme.yoga.colors),
        )}`,
      );

    return (
      <Item>
        <Wrapper>
          {icon && (
            <IconWrapper>
              {isValidElement(icon) ? (
                icon
              ) : (
                <Icon as={icon} size="small" fill={variant || 'text.primary'} />
              )}
            </IconWrapper>
          )}
          <ItemText color={itemColor}>{text}</ItemText>
        </Wrapper>
        {Boolean(Object.keys(buttonProps).length) && (
          <TouchableWithoutFeedback {...buttonProps}>
            <Button>
              <ButtonText>{children}</ButtonText>
            </Button>
          </TouchableWithoutFeedback>
        )}
      </Item>
    );
  },
);

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: string.isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
  /** if provided a color variant, like "vibin", "hope", "energy" the icon and
   * the item text will be rendered on this color.
   */
  variant: string,
};

export { List, ListItem };
