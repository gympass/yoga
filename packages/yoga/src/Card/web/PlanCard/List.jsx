import React, { isValidElement } from 'react';
import styled, { css, withTheme } from 'styled-components';
import { string, node, shape, oneOfType, func, bool } from 'prop-types';

import Text from '../../../Text';
import theme from '../../../Theme/helpers/themeReader';
import Box from '../../../Box';

const { card, cardweb } = theme.components;

const truncateStyle = css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const List = styled.ul`
  margin: ${cardweb.plan.list.margin.top}px 0 0;
  padding: 0;

  list-style: none;
`;

const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-right: ${cardweb.plan.list.item.icon.margin.right}px;
  min-height: ${props => props.theme.yoga.spacing.medium}px;
`;

const Item = styled.li`
  :not(:last-child) {
    margin-bottom: ${card.plan.list.item.margin.bottom}px;
  }
`;

const Wrapper = styled(Box)`
  display: flex;
  flex-direction: column;
  width: 100%;

  ${props =>
    props.as === 'button' &&
    css`
      border: none;
      padding: 0;
      background: transparent;
      cursor: pointer;
      text-align: start;
    `}

  svg {
    flex-shrink: 0;
  }
`;

const ItemText = styled(Text.Small)`
  display: inline-block;
  height: 100%;
  vertical-align: middle;

  color: ${card.plan.list.item.font.color};

  ${props => props.truncate && truncateStyle}
`;

const Button = styled.button`
  display: block;
  padding: 0;
  margin-top: ${card.plan.list.button.margin.top}px;

  width: 100%;

  letter-spacing: normal;

  background-color: transparent;
  border: none;

  font-family: ${theme.baseFont.family};
  font-size: ${card.plan.list.button.font.size}px;
  font-weight: ${card.plan.list.button.font.weight};
  color: ${card.plan.list.button.font.color};

  cursor: pointer;
  text-decoration: none;
  text-align: left;
  outline: none;

  ${truncateStyle}
`;

const emptyObj = {};

const ListItem = withTheme(props => {
  const {
    text,
    truncate = true,
    icon: Icon,
    buttonProps = emptyObj,
    theme: yogaTheme,
    children,
    onClick,
  } = props;
  const wrapperProps = onClick ? { as: 'button', type: 'button', onClick } : {};

  return (
    <Item>
      <Wrapper {...wrapperProps}>
        <Box display="flex" alignItems="flex-start" width="100%">
          {Icon && (
            <IconWrapper>
              {isValidElement(Icon) ? (
                Icon
              ) : (
                <Icon
                  width={16}
                  height={16}
                  fill={yogaTheme.yoga.colors.text.primary}
                />
              )}
            </IconWrapper>
          )}
          <ItemText as="span" truncate={truncate}>
            {text}
          </ItemText>
        </Box>
        {children}
      </Wrapper>
      {Boolean(Object.keys(buttonProps).length) && <Button {...buttonProps} />}
    </Item>
  );
});

List.displayName = 'PlanCard.List';
ListItem.displayName = 'PlanCard.ListItem';
Button.displayName = 'PlanCard.ListButton';

ListItem.propTypes = {
  text: oneOfType([string, node]).isRequired,
  /** an icon to be displayed on the begin of the item */
  icon: oneOfType([node, func]),
  /** if provided displays a button below the item text. It accepts all button
   * element props */
  buttonProps: shape({}),
  /** if provided makes the item clickable */
  onClick: func,
  children: node,
  truncate: bool,
};

export { List, ListItem };
