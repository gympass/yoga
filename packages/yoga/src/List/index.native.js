import { List, ListItem } from './native';

const ExportList = Object.assign(List, { Item: ListItem });

export default ExportList;
