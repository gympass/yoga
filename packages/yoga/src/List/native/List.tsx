import React from 'react';
import type { FlatListProps } from 'react-native';
import type { FlatList } from 'react-native';
import styled from 'styled-components/native';

const StyledFlatList = styled.FlatList`
  width: 100%;
`;

/** Lists are a continuous group of text or images. They are composed of items
containing primary and supplemental actions, which are represented by icons and
text. */
const List = React.forwardRef<FlatList, FlatListProps<any>>(
  ({ ...rest }, ref) => {
    return <StyledFlatList ref={ref} {...rest} />;
  },
);

List.displayName = 'List';

export default List;
