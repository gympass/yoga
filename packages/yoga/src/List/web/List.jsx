import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledList = styled.ul`
  display: flex;

  width: 100%;
  padding: 0;
  margin: 0;

  list-style: none;

  ${({
    divided,
    horizontal,
    theme: {
      yoga: {
        components: { list },
      },
    },
  }) => `
    flex-direction: ${horizontal ? 'row' : 'column'};

    > * {
        ${
          divided
            ? `border-${horizontal ? 'right' : 'bottom'}: ${
                list.border.width
              }px solid ${list.border.color}`
            : ''
        };

        &:last-child {
          border: none;
        }
      }
    }
  `}
`;

/** Lists are a continuous group of text or images. They are composed of items
containing primary and supplemental actions, which are represented by icons and
text. */

const List = React.forwardRef(
  ({ horizontal = false, divided = true, ...rest }, ref) => (
    <StyledList horizontal={horizontal} divided={divided} ref={ref} {...rest} />
  ),
);

List.propTypes = {
  horizontal: PropTypes.bool,
  divided: PropTypes.bool,
};

List.displayName = 'List';

export default List;
