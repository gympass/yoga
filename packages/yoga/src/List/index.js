import { List, ListItem, LinkItem } from './web';

const ExportList = Object.assign(List, { Item: ListItem, LinkItem });

export default ExportList;
