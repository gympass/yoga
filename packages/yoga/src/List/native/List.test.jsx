import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import ThemeProvider from '../../Theme';
import List from '..';

const data = [
  { key: 'Devin' },
  { key: 'Dan' },
  { key: 'Dominic' },
  { key: 'Jackson' },
  { key: 'James' },
  { key: 'Joel' },
  { key: 'John' },
  { key: 'Jillian' },
  { key: 'Jimmy' },
  { key: 'Julie' },
];

describe('<List />', () => {
  describe('Snapshots', () => {
    describe('Without props', () => {
      it('should match snapshot with a default list', () => {
        const { toJSON } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
      });

      it('should match snapshot with a small list', () => {
        const { toJSON } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item small>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
      });

      it('should match snapshot with a list without divisors', () => {
        const { toJSON } = render(
          <ThemeProvider>
            <List
              data={data}
              renderItem={({ item }) => (
                <List.Item divided={false}>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
      });

      it('should match snapshot with a horizontal list', () => {
        const { toJSON } = render(
          <ThemeProvider>
            <List
              horizontal
              data={data}
              renderItem={({ item }) => (
                <List.Item>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
      });

      it('should call onPress when selectable item is pressed', () => {
        const a = { key: 'Devin', onPress: jest.fn() };
        const { getByText } = render(
          <ThemeProvider>
            <List
              data={[a]}
              renderItem={({ item }) => (
                <List.Item onPress={item.onPress}>
                  <Text>{item.key}</Text>
                </List.Item>
              )}
            />
          </ThemeProvider>,
        );

        const selectableItem = getByText(a.key);

        fireEvent.press(selectableItem);
        expect(a.onPress).toHaveBeenCalled();
      });
    });
  });
});
