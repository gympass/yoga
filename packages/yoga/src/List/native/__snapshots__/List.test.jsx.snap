// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Snapshots Without props should match snapshot with a default list 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "key": "Devin",
      },
      Object {
        "key": "Dan",
      },
      Object {
        "key": "Dominic",
      },
      Object {
        "key": "Jackson",
      },
      Object {
        "key": "James",
      },
      Object {
        "key": "Joel",
      },
      Object {
        "key": "John",
      },
      Object {
        "key": "Jillian",
      },
      Object {
        "key": "Jimmy",
      },
      Object {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a horizontal list 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "key": "Devin",
      },
      Object {
        "key": "Dan",
      },
      Object {
        "key": "Dominic",
      },
      Object {
        "key": "Jackson",
      },
      Object {
        "key": "James",
      },
      Object {
        "key": "Joel",
      },
      Object {
        "key": "John",
      },
      Object {
        "key": "Jillian",
      },
      Object {
        "key": "Jimmy",
      },
      Object {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        divided={true}
        small={false}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a list without divisors 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "key": "Devin",
      },
      Object {
        "key": "Dan",
      },
      Object {
        "key": "Dominic",
      },
      Object {
        "key": "Jackson",
      },
      Object {
        "key": "James",
      },
      Object {
        "key": "Joel",
      },
      Object {
        "key": "John",
      },
      Object {
        "key": "Jillian",
      },
      Object {
        "key": "Jimmy",
      },
      Object {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={false}
        small={false}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 20,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<List /> Snapshots Without props should match snapshot with a small list 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "key": "Devin",
      },
      Object {
        "key": "Dan",
      },
      Object {
        "key": "Dominic",
      },
      Object {
        "key": "Jackson",
      },
      Object {
        "key": "James",
      },
      Object {
        "key": "Joel",
      },
      Object {
        "key": "John",
      },
      Object {
        "key": "Jillian",
      },
      Object {
        "key": "Jimmy",
      },
      Object {
        "key": "Julie",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Devin
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Dan
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Dominic
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jackson
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          James
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Joel
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          John
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jillian
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Jimmy
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        divided={true}
        small={true}
        style={
          Array [
            Object {
              "borderBottomColor": "#F5F5FA",
              "borderBottomWidth": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            },
          ]
        }
      >
        <Text>
          Julie
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
